src/doc/trpl/compound-data-types.md

```diff
@@ -199,154 +199,116 @@ destructuring `let`.
 ## Enums
 
 Finally, Rust has a "sum type", an *enum*. Enums are an incredibly useful
-feature of Rust, and are used throughout the standard library. This is an enum
-that is provided by the Rust standard library:
+feature of Rust, and are used throughout the standard library. An `enum` is
+a type which ties a set of alternates to a specific name. For example, below
+we define `Character` to be either a `Digit` or something else. These
+can be used via their fully scoped names: `Character::Other`.
 
 ```{rust}
-enum Ordering {
-    Less,
-    Equal,
-    Greater,
+enum Character {
+    Digit(i32),
+    Other,
 }
 ```
 
-An `Ordering` can only be _one_ of `Less`, `Equal`, or `Greater` at any given
-time.
+An `enum` variant, can be defined as most normal types. Below some example types
+have been listed which also would be allowed in an `enum`.
 
-Because `Ordering` is provided by the standard library, we can use the `use`
-keyword to use it in our code. We'll learn more about `use` later, but it's
-used to bring names into scope.
+```rust
+struct Empty;
+struct Color(i32, i32, i32);
+struct Length(i32);
+struct Status { Health: i32, Mana: i32, Attack: i32, Defense: i32 }
+struct HeightDatabase(Vec<i32>);
+```
 
-Here's an example of how to use `Ordering`:
+So you see that depending on the sub-datastructure, the `enum` variant, same as a
+struct variant, may or may not hold data. That is, in `Character`, `Digit` is a name
+tied to an `i32` where `Other` is just a name. However, the fact that they are distinct
+makes this very useful.
 
-```{rust}
-use std::cmp::Ordering;
+As with structures, enums don't by default have access to operators such as
+compare ( `==` and `!=`), binary operations (`*` and `+`), and order
+(`<` and `>=`). As such, using the previous `Character` type, the
+following code is invalid:
 
-fn cmp(a: i32, b: i32) -> Ordering {
-    if a < b { Ordering::Less }
-    else if a > b { Ordering::Greater }
-    else { Ordering::Equal }
-}
+```{rust, norun}
```

Comment:
I don't remember how you tell rustdoc not to test this example...I'll look it up later